internal/notify/notifications.lua

```diff
@@ -267,6 +267,79 @@ local function get_stranded_message()
     end
 end
 
+local function get_blood()
+    return dfhack.world.getAdventurer().body.blood_count
+end
+
+local function get_max_blood()
+    return dfhack.world.getAdventurer().body.blood_max
+end
+
+local function get_max_breath()
+    local adventurer = dfhack.world.getAdventurer()
+    local toughness = dfhack.units.getPhysicalAttrValue(adventurer, df.physical_attribute_type.TOUGHNESS)
+    local endurance = dfhack.units.getPhysicalAttrValue(adventurer, df.physical_attribute_type.ENDURANCE)
+    local base_ticks = 200
+
+    return math.floor((endurance + toughness) / 4) + base_ticks
+end
+
+local function get_breath()
+    return get_max_breath() - dfhack.world.getAdventurer().counters.suffocation
+end
+
+local function get_suffocation_message()
+    if get_breath() < get_max_breath() then
+        local label_text = {}
+        local suffocation_pen = COLOR_LIGHTCYAN
+        -- if self.blink then
+        --     suffocation_pen = COLOR_CYAN
+        -- end
+        table.insert(label_text, {text = "Suffocating!", pen = suffocation_pen, width = 14})
+
+        local bar_width = 16
+        local percentage = get_breath() / get_max_breath()
+        local barstop = math.floor((bar_width * percentage) + 0.5)
+        for idx = 0, bar_width-1 do
+            local color = COLOR_LIGHTCYAN
+            local char = 219
+            if idx >= barstop then
+                -- offset it to the hollow graphic
+                color = COLOR_DARKGRAY
+                char = 177
+            end
+            table.insert(label_text, { width = 1, tile={ch=char, fg=color}})
+        end
+        return label_text
+    end
+end
+
+local function get_bleeding_message()
+    if get_blood() < get_max_blood() then
+        local label_text = {}
+        local bloodloss_pen = COLOR_RED
+        -- if self.blink then
+        --     bloodloss_pen = COLOR_LIGHTRED
+        -- end
+        table.insert(label_text, {text = "Bloodloss!", pen = bloodloss_pen, width = 14})
+
+        local margin = 14
+        local percentage = get_blood() / get_max_blood()
+        local barstop = math.floor((margin * percentage) + 0.5)
+        for idx = 0, margin-1 do
+            local color = COLOR_RED
+            local char = 219
+            if idx >= barstop then
+                -- offset it to the hollow graphic
+                color = COLOR_DARKGRAY
+                char = 177
+            end
+            table.insert(label_text, { width = 1, tile={ch=char, fg=color}})
+        end
```

Comment:
please factor out this common logic. consider combining `get_suffocation_message` and `get_bleeding_message` into one function and using `curry` to pass in parameters